widgets/htdocs/alphafold/exonsControlPanel.js

```diff
@@ -0,0 +1,170 @@
+        font-weight: bold;
+        padding: 8px 4px 8px 6px;
+      }
+
+      .exons-count {
+        min-width: 10px;
+        padding: 2px 6px;
+        font-size: 10px;
+        font-weight: bold;
+        border: 1px solid var(--main-dark);
+        text-align: center;
+        border-radius: 10px;
+      }
+
+      .show-feature, .panel-content-toggle {
+        width: 16px;
+        height: 16px;
+        border: none;
+        outline: none;
+        cursor: pointer;
+      }
+
+      .show-feature_hidden {
+        background: url(/i/16/eye_no.png);
+      }
+
+      .show-feature_visible {
+        background: url(/i/16/eye.png);
+      }
+
+      .toggle-expanded {
+        background: url(/i/16/minus-button.png) no-repeat right center;
+      }
+
+      .toggle-collapsed {
+        background: url(/i/16/plus-button.png) no-repeat right center;      
+      }
+
+      .exon-details-wrapper {
+        padding: 5px 5px 5px 10px;
+      }
+
+      .exon-details {
+        width: 100%;
+        border: 1px solid #ccc;
+      }
+
+      .exon-details th {
+        background-color: #ccc;
+        padding: 4px;
+      }
+
+      .exon-details tbody tr td:nth-child(2),
+      .exon-details tbody tr td:nth-child(3) {
+        text-align: center;
+      }
+
+      .exon-row td:first-child {
+        border-left-width: 5px;
+        border-left-style: solid;
+        padding-left: 6px;
+      }
+    `;
+  }
+
+  static get properties() {
+    return {
+      exons: { attribute: false },
+      selectedExonIndices: { attribute: false },
+      isExpanded: { state: true }
+    }
+  }
+
+  areAllExonsVisible() {
+    return this.selectedExonIndices.length === this.exons.length;
+  }
+
+  isExonVisible(exonIndex) {
+    return this.selectedExonIndices.includes(exonIndex);
+  }
+
+  toggleSelectedExon(exonIndex) {
+    if (this.selectedExonIndices.includes(exonIndex)) {
+      const filteredIndices = this.selectedExonIndices.filter(index => index !== exonIndex);
+      this.onExonSelectionChange(filteredIndices);
+    } else {
+      this.onExonSelectionChange([...this.selectedExonIndices, exonIndex]);
+    }
+  }
+
+  toggleAllExons() {
+    if (this.selectedExonIndices.length) {
+      this.onExonSelectionChange([]);
+    } else {
+      const exonIndices = this.exons.map((_, index) => index);
+      this.onExonSelectionChange(exonIndices);
+    }
+  }
+
+  render() {
+    return html`
+      <control-panel title="Exons">
+        <div class="panel-summary">
+          Exons
+          <span class="exons-count">${this.exons.length}</span>
+          <button
+            class="show-feature ${this.areAllExonsVisible() ? 'show-feature_visible' : 'show-feature_hidden'}"
+            @click=${this.toggleAllExons.bind(this)}
+          ></button>
+          <button
+            class="panel-content-toggle ${this.isExpanded ? 'toggle-expanded' : 'toggle-collapsed'}"
+            @click=${() => this.isExpanded = !this.isExpanded}
```

Comment:
doubt, can you please tell me why @click is used instead of onClick?

Reply:
This is the templating syntax of the library that I am using 😃 Remember, it's not JSX and React; it is lit-html and Lit. Here's the [relevant part of documentation](https://lit.dev/docs/templates/expressions/#event-listener-expressions).

Reply:
ok cool. I was getting confused with both the syntaxes. Thanks